src/components/ha-assist-chat.ts

```diff
@@ -435,28 +475,72 @@ export class HaAssistChat extends LitElement {
     this._processing = true;
     this._audio?.pause();
     this._addMessage({ who: "user", text });
-    const message: AssistMessage = {
+    let hassMessage = {
       who: "hass",
       text: "…",
+      error: false,
     };
+    let currentDeltaRole = "";
     // To make sure the answer is placed at the right user text, we add it before we process it
-    this._addMessage(message);
+    this._addMessage(hassMessage);
     try {
       const unsub = await runAssistPipeline(
         this.hass,
         (event) => {
+          if (event.type === "intent-progress") {
+            const delta = event.data.chat_log_delta;
+
+            // new message and previous message has content
+            if (delta.role) {
+              // If currentDeltaRole exists, it means we're receiving our
+              // second or later message. Let's add it to the chat.
+              // Reuse the previous message if it had no content
```

Comment:
I can't find where "if it had no content" is happening.

I read this as this block is checking if the existing message has content and appending to it but otherwise dropping the final delta. Is the point here that if there is already content then a delta with an assistant role is empty and ends the message?

Reply:
If the previous message had no content that we shown, the content of `hassMessage` is still equal to `…`. So we check that on line 501:

```ts
hassMessage.text !== "…"
```

Reply:
I feel i am missing something silly:
- If it is "..." then it has no content.
- If it is not "..." then it has content.

Reply:
Right. So if a new message comes in, and the previous message no longer just contains `…`, some content was added. In that case, we add a second message to the log and start adding the deltas to that one. You can see that happen in the video above, where a 2nd "role" comes in, and it starts a new chat bubble.

Reply:
OK i think removing "Reuse the previous message if it had no content" would be helpful since its covered by the comment "new message and previous message has content". That comment implies to me that its part of the if statement, but its not, its part of the fact that the delta is arriving.